### IS362_Project_4
Redjol Olli

#### Project 4 – Predictive Analysis using scikit-learnYour assignment is to:
- Start with the mushroom data in the pandas DataFramethat you constructed in your “Assignment – Preprocessing Data with sci-kit learn.”
- Usescikit-learnto determine which of the two predictor columns that you selected (odor and one other column of your choice) most accurately predicts whether or not a mushroom is poisonous.  There is an additional challenge here—to use scikit-learn’s predictive classifiers, you’ll want to convert each of your two (numeric categorical) predictor columns into a set of columns.  See for one approach pandas get_dummies() method. 
- Clearly state your conclusions along with any recommendations for further analysis.


In [1]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt
import seaborn as sns
%matplotlib inline

from sklearn.linear_model import LinearRegression
from sklearn.neighbors import KNeighborsClassifier
from sklearn.linear_model import LogisticRegression
from sklearn import linear_model


In [2]:
df = pd.read_csv('https://archive.ics.uci.edu/ml/machine-learning-databases/mushroom/agaricus-lepiota.data',sep=',', 
                         header=None, 
                         usecols=[0,3,5,9], 
                         names=["Edible","Cap Color","Odor","Gill Color"])

df.replace(to_replace={"Edible":{'p': 1, 'e': 0}}, inplace=True)
df.replace(to_replace={"Cap Color":{'n':0, 'b':1, 'c':2, 'g':3, 'r':4, 'p':5, 'u':6, 'e':7, 'w':8, 'y':9}}, inplace=True)
df.replace(to_replace={"Odor":{'a':0, 'l':1, 'c':2, 'y':3, 'f':4, 'm':5, 'n':6, 'p':7, 's':8}}, inplace=True)
df.replace(to_replace={"Gill Color":{'k':0, 'n':1, 'b':2, 'h':3, 'g':4, 'r':5, 'o':6, 'p':7, 'u':8, 'e':9, 'w':10, 'y':11}}, inplace=True)

df.head()

Edible  Cap Color  Odor  Gill Color
0       1          0     7           0
1       0          9     0           0
2       0          8     1           1
3       1          8     7           1
4       0          3     6           0

## Determining total number of edible mushroom and  poisonous.

In [3]:
df.groupby('Edible').count()

Cap Color  Odor  Gill Color
Edible                             
0            4208  4208        4208
1            3916  3916        3916

Total number of edible mushrooms in this database is 4208, and 3916 of them are poisonous. 

In [4]:
df.groupby(['Edible', 'Odor']).count()

Cap Color  Gill Color
Edible Odor                       
0      0           400         400
       1           400         400
       6          3408        3408
1      2           192         192
       3           576         576
       4          2160        2160
       5            36          36
       6           120         120
       7           256         256
       8           576         576

There are 3 edible mushrooms, odors 0, 1, 6. Also, odor 6 is shown in the poisonous mushrooms.

## Deterimining mushrooms edibility by Cap Color.

In [6]:
df.groupby(['Edible', 'Cap Color']).count()

Odor  Gill Color
Edible Cap Color                  
0      0          1264        1264
       1            48          48
       2            32          32
       3          1032        1032
       4            16          16
       5            56          56
       6            16          16
       7           624         624
       8           720         720
       9           400         400
1      0          1020        1020
       1           120         120
       2            12          12
       3           808         808
       5            88          88
       7           876         876
       8           320         320
       9           672         672

Mushrooms with a Cap Color of 4 or 6 are edible, and no mushroom with these colors is shown on poisonous category.

## Deterimining mushrooms edibility by Gill Color.

In [8]:
df.groupby(['Edible', 'Gill Color']).count()

Cap Color  Odor
Edible Gill Color                 
0      0                 344   344
       1                 936   936
       3                 204   204
       4                 248   248
       6                  64    64
       7                 852   852
       8                 444   444
       9                  96    96
       10                956   956
       11                 64    64
1      0                  64    64
       1                 112   112
       2                1728  1728
       3                 528   528
       4                 504   504
       5                  24    24
       7                 640   640
       8                  48    48
       10                246   246
       11                 22    22

Mushrooms with a Gill Color of 6 and 9 always be edible.
Mushrooms with a Gill Color of 3 and 5 always be poisonous.

## Conclusion


Based on the finding the "odor" category will be the most reliable method of predicting mushrooms edibility.